### PR TITLE
CDAP-2464 Move Documentation that highlights different CDAP modes

### PR DESCRIPTION
**Pull request by awholegunch**
Move "mode" info from "Installation" to "Developers' Manual". Fixing a few minor issues along the way.

Build is broken because ``release/cdap-3.3-compatible`` branches in all the other repos that the docs depend on do not yet exist.

Passes [Quick Build No. 8](http://builds.cask.co/browse/CDAP-DOB69-8)

**Pages of Interest:**
[Modes and Components](http://builds.cask.co/artifact/CDAP-DOB69/shared/build-8/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/overview/modes.html)

Replacement for https://github.com/caskdata/cdap/pull/4658
Fix for https://issues.cask.co/browse/CDAP-2464

### CODE REVIEW

File: cdap-docs/developers-manual/source/overview/modes.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+.. _standalone-data-application-platform:
+
+.. rubric:: Standalone CDAP
+
+The **Standalone CDAP** allows you to run the entire CDAP stack in a single Java Virtual
+Machine on your local machine and includes a local version of the CDAP UI. The
+underlying Big Data infrastructure is emulated on top of your local file system. All data
+is persisted.
+
+The Standalone CDAP by default binds to the localhost address, and is not available for
```
Comment (anwar6953) [inline cdap-docs/developers-manual/source/overview/modes.rst]: I'm not sure this is true; a Standalone instance may be available for remote access (from another machine).

File: cdap-docs/vars [hunk 2]
```diff
@@ -19,5 +19,4 @@ WITH_JAVADOCS="$WITHOUT"
 MANUALS="introduction developers-manual cdap-apps admin-manual integrations examples-manual reference-manual faqs"
 
 # This needs to be explicitly set as Git has trouble identifying it.
-GIT_BRANCH_TYPE="develop"
-GIT_BRANCH_PARENT="develop"
+GIT_BRANCH_PARENT="release/3.3"
```
Comment (anwar6953) [inline cdap-docs/vars]: just wondering - this will need to change each time we release CDAP, right?

File: cdap-docs/_common/_source/index.rst [hunk 3]
```diff
@@ -60,12 +60,12 @@ on the installation, monitoring and diagnosing fully distributed CDAP in a Hadoo
...
 
   - **Getting Started Developing:** A quick, hands-on introduction to developing with CDAP
-  - **Overview:** The overall architecture and technology behind CDAP
+  - **Overview:** The overall architecture, abstractions, modes, and components behind CDAP
   - **Building Blocks:** The two core abstractions in CDAP: *Data* and *Applications*, and their components
   - **Security:** Perimeter security, configuration and client authentication
   - **Testing and Debugging:** Test framework plus tools and practices for debugging your applications
   - **Ingesting Data:** Different techniques for ingesting data into CDAP
-  - **Advanced Topics:** Best practices for CDAP development, adding a custom logback
+  - **Advanced Topics:** Class loading in CDAP, best practices for CDAP development, adding a custom logback
```
Comment (jwang47) [inline cdap-docs/_common/_source/index.rst]: Maybe put "Class loading in CDAP" after "best practices"

File: cdap-docs/vars (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/vars]: It needs to be the parent branch; once that is set, it doesn't change for that line. Each time you cut a new release branch is when it happens. Unfortunately, I've found no other way of doing this...

File: cdap-docs/_common/_source/index.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/_common/_source/index.rst]: Re-arranged order (three places).

File: cdap-docs/developers-manual/source/overview/modes.rst (hunk 1 above)
Comment (jwang47) [inline cdap-docs/developers-manual/source/overview/modes.rst]: We can just remove this part ("not available for ..")

Comment (awholegunch) [inline cdap-docs/developers-manual/source/overview/modes.rst]: Removed the entire sentence, as it no longer is relevant or useful.

**Comment by awholegunch** [comment]
@anwar6953 @jwang47 I've updated this and started a new Quick Build...please take another look.

**Comment by jwang47** [comment]
LGTM

**Pull request merged**